src/util/helm.js

```diff
@@ -1,23 +1,44 @@
-const execa = require("execa");
+const { createHash } = require("crypto");
 const path = require("path");
 const fs = require("fs/promises");
+const execa = require("execa");
 const YAML = require("yawn-yaml/cjs");
 
-const extract_fs = require("fs");
-const extract_YAML = require('yaml');
+const helmDependencyRepos = [];
 
 const helmVersion = async () => {
     const { stdout: version } = await execa("helm", [
         "version",
-        "--template='{{.Version}}'"
+        "--template='{{.Version}}'",
     ]);
 
     return version;
 };
 
 const helmLint = (chart) => execa("helm", ["lint", chart]);
 
-const helmRepoAdd = (repo, name) => execa("helm", ["repo", "add", name, repo]);
+// given a chart path, find this chart's dependencies, and add all the owning
+// chart repositories via `helm repo add ${repository} ${url}`.
+// this is a prerequisite for `helm dependency build`.
+const helmRepoAddDependencies = async (chartPath) => {
+    const chartYamlFile = path.join(chartPath, "Chart.yaml");
+    const chartYaml = await fs.readFile(chartYamlFile);
+
+    const doc = new YAML(chartYaml.toString());
+
+    await Promise.all(doc.json.dependencies.map(async (dependency) => {
+        // the name doesn't matter, in fact it's better that this doesn't possibly conflict with existing helm repos
+        const name = createHash("sha1").update(dependency.repository).digest("hex");
```

Comment:
Do you think there might be any weird side effects in the future of naming helm repo add using a sha hash?